docs/resources/oceancd_rollout_spec.md

```diff
@@ -0,0 +1,191 @@
+ 
+  traffic {
+    canary_service = "canary-service-test"
+    stable_service = "stable-service-test"
+    
+    alb {
+      alb_annotation_prefix  = "custom-ingress-kubernates"
+      alb_ingress            = "test-ingress" 
+      alb_root_service       = "test-root-service"
+      service_port           = 8080
+      stickiness_config {
+            duration_seconds = 3600
+            enabled          = true
+        }
+      } 
+    }
+    
+    ambassador {
+        mappings = ["test-mappong","test2-mapping"]    
+    }
+    
+    istio {
+        destination_rule {
+            canary_subset_name    = "canary-subset"
+            destination_rule_name = "test-dest-rule"
+            stable_subset_name    = "stable-subset"
+        }
+        virtual_services {
+            virtual_service_name   = "test-name"
+            virtual_service_routes = ["route1","route2"]
+            tls_routes {
+                port      = 80
+                sni_hosts = "spot.io"
+            }    
+        }
+    }
+    
+    nginx {
+        nginx_annotation_prefix = "customIngress.nginx"
+        stable_ingress          = "hello-ingress"
+        additional_ingress_annotation {
+            canary_by_header = "x-canary-test"
+            key1             = "test"
+        }
+    }
+    
+    ping_pong {
+        ping_service = "stable-service"
+        pong_service = "canary-service"
+    }
+    
+    smi {
+        smi_root_service   = "stable-service"
+        traffic_split_name = "rollout-expample-traffic-split"
+    }   
+  }   
+  //------------------------------------------------------------------------------    
+}    
+    
+```
+
+```
+output "name" {
+  value = spotinst_oceancd_rollout_spec.example.name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `rollout_spec_name` - (Required) Identifier name for Ocean CD Rollout Spec. Must be unique
+* `failure_policy` - (Optional) Holds information on how to react when failure happens.
+    * `action` - (Required) Choose an action to perform on failure. Default is `abort`.  Enum: "abort" "pause" "promote".
+* `spot_deployment` - (Optional) Represents the SpotDeployment selector.
+    * `cluster_id` - (Optional) Ocean CD cluster identifier for the references `SpotDeployment`.
+    * `spot_deployment_name` - (Optional) The name of the `SpotDeployment` resource
+    * `namespace` - (Optional) The namespace which the `SpotDeployment` resource exists within.
+* `spot_deployments` - (Optional) You must specify either `spotDeployment` OR `spotDeployments` but not both. Every SpotDeployment has to be unique. If more than one `SpotDeployment` has been configured, no `traffic` managers can be set as part of the RolloutSpec.For such case ensure that each of the chosen SpotDeployments are being exposed with different Kubernetes services.
+    * `cluster_id` - (Optional) Ocean CD cluster identifier for the references `SpotDeployment`.
```

Comment:
modify the attribute names according to the consts file

Reply:
modified